controllers/create_pgdeployment.go

```diff
@@ -0,0 +1,144 @@
+// Copyright Contributors to the Open Cluster Management project
+package controllers
+
+import (
+	"context"
+
+	cachev1 "github.com/stolostron/search-v2-operator/api/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *OCMSearchReconciler) createPGDeployment(request reconcile.Request,
+	deploy *appsv1.Deployment,
+	instance *cachev1.OCMSearch,
+) (*reconcile.Result, error) {
+
+	found := &appsv1.Deployment{}
+	err := r.Get(context.TODO(), types.NamespacedName{
+		Name:      deploy.Name,
+		Namespace: request.Namespace,
+	}, found)
+	if err != nil && errors.IsNotFound(err) {
+
+		err = r.Create(context.TODO(), deploy)
+		if err != nil {
+			return &reconcile.Result{}, err
+		} else {
+			return nil, nil
+		}
+	} else if err != nil {
+		return &reconcile.Result{}, err
+	}
+
+	return nil, nil
+}
+
+func (r *OCMSearchReconciler) PGDeployment(instance *cachev1.OCMSearch) *appsv1.Deployment {
+
+	deploymentLabels := map[string]string{
+		"name": "search-postgres",
+	}
+	deployment := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "search-postgres",
+			Namespace: instance.Namespace,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: deploymentLabels,
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: deploymentLabels,
+				},
+			},
+		},
+	}
+	postgresContainer := corev1.Container{
+		Name:  "search-posgres",
+		Image: "registry.redhat.io/rhscl/postgresql-13-rhel7:1-31.1638430360",
```

Comment:
We'll need an easy way to update this image tag later.  I'm sure there will be CVEs.

Reply:
Right - We will need to parametrize in the deployment for this and all search components .